src/include/mpir_request.h

```diff
@@ -387,4 +387,26 @@ int MPIX_Grequest_start_impl(MPI_Grequest_query_function *,
                              MPIX_Grequest_poll_function *,
                              MPIX_Grequest_wait_function *, void *, MPIR_Request **);
 
+/* Check for extended generalized requests. NOTE: wait_fn is always initialized with poll_fn. */
+MPL_STATIC_INLINE_PREFIX int MPIX_Request_is_ex_generalized(MPIR_Request * request_ptr)
```

Comment:
Oh one more thing. These can be `MPIR_` namespace. `MPIX_` is intended for functions exposed to the user but not yet part of the MPI standard.